pkg/controller/servicemesh/controlplane/reconciler.go

```diff
@@ -597,3 +601,43 @@ func componentFromChartName(chartName string) string {
 	_, componentName := path.Split(chartName)
 	return componentName
 }
+
+func overrideDiscoverySelectors(spec *v2.ControlPlaneSpec, controlPlaneNamespace string) {
+	var discoverySelectors []*metav1.LabelSelector
+	if spec.MeshConfig != nil {
+		discoverySelectors = spec.MeshConfig.DiscoverySelectors
+	}
+	if discoverySelectors == nil {
+		return
+	}
+	for _, s := range discoverySelectors {
+		if s.MatchExpressions != nil {
```

Comment:
What about `s.MatchLabels`? Don't we need to check those as well?

Reply:
We would have to know namespace labels - this is easy during reconciliation, but what about removing relevant label from the namespace? Then we would have to watch namespace and add proper `matchExpressions` on a change. That's too complicated, so I think we can just add single `matchExpressions` rule for the namespace name. If `matchLabels` matches the control plane namespace and `matchExpressions` as well - that's fine. Rules won't be optimal, but still correct.